SocketRocket/ProxyConnect.m

```diff
@@ -0,0 +1,468 @@
+#import "ProxyConnect.h"
+#import "SRError.h"
+#import "NSRunLoop+SRWebSocket.h"
+
+static inline void ProxyFastLog(NSString *format, ...);
+
+typedef void (^connectDoneBlock_t)(NSError * error, NSInputStream *readStream, NSOutputStream *writeStream);
+
+@interface ProxyConnect() <NSStreamDelegate>
+
+@property (nonatomic, strong) NSURL *url;
+@property (nonatomic, strong) NSInputStream * inputStream;
+@property (nonatomic, strong) NSOutputStream * outputStream;
+
+@end
+@implementation ProxyConnect {
+    
+    connectDoneBlock_t  connectDoneHandler;
+    
+    NSString *_httpProxyHost;
+    uint32_t _httpProxyPort;
+    
+    CFHTTPMessageRef _receivedHTTPHeaders;
+
+    NSString *_socksProxyHost;
+    uint32_t _socksProxyPort;
+    NSString *_socksProxyUsername;
+    NSString *_socksProxyPassword;
+    
+    BOOL _secure;
+    
+    NSMutableArray<NSData *> *_inputQueue;
+    NSOperationQueue * _writeQueue;
+    
+}
+
+-(instancetype)initWithURL:(NSURL *)url
+{
+    if ([super init]) {
```

Comment:
This would be `self = [super init]` or the common one we use:
```
self = [super init];
if (!self) return self;
```
which will also avoid creating an extra indentation level.